### PR TITLE
React and Redux Group Project: Space Travelers Hub

### PR DESCRIPTION
**Pull request by rmiyoyo**
# Work Done

## Description

In this task, we worked with the real live data from the SpaceX API. We built a web application for a company that provides commercial and scientific space travel services. The application allows users to book rockets and join selected space missions.

#### Rockets /Dragons

The Rockets section displays a list of all available SpaceX rockets. Users can book each rocket by clicking the reservation button or cancel the previously made booking. The same layout is used to form the Dragons section.

#### Missions

The Missions section displays a list of current missions along with their brief description and participation status. There is also a button next to each mission that allows users to join the selected mission or leave the mission the user joined earlier.

#### My Profile

The My Profile section displays all reserved rockets and space missions.

#### Config & basic setup

- Created React App.
- Installed [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/articles/react_router.md).
- Downloaded the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
- Created **routes** and **view** components: `rockets`, `missions`, `my profile`, `dragons`.
  - Used `<NavLink />` for the page navigation links and style active class to indicate which section/page user is currently on (underline active navigation link).
- Created directories for all Redux state slice files:`rockets`, `missions`, `dragons`.
- Fetched data and updated Redux store
- Once the data are fetched, dispatched an action to store the selected data in Redux store.
- Used `useSelector()` Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes. 
- Wrote actions and reducers for booking rockets/dragons and joining missions.
- Created a reducer and action dispatcher for the "Join Mission" button. 
- Wrote actions and reducers for canceling rockets/dragons and leaving missions.
- Rockets that have already been reserved show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve rocket".
- Dragons that have already been reserved show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve dragon".
- Missions that the user has joined already show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button.
- Rockets/Dragons and Missions use the React conditional rendering syntax.
- Composed three column layout and list ONLY the rockets/dragons reserved and missions joined by the user.

### CODE REVIEW

File: src/components/Navbar.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import '../css/Navbar.css';
+import logo from '../assets/planet.png';
```
Comment (algerina) [inline src/components/Navbar.jsx:4]: - [ ] As required <NavLink /> should be used for navigation links.

File: src/components/MissionsComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                    >
+                      { joined ? 'Leave Mission' : 'Join Mission'}
+                    </button>
+                  </td>
+                </tr>
+              );
+            })
+          }
+        </tbody>
+      </table>
```
Comment (algerina) [inline src/components/MissionsComponent.jsx:56]: - [ ] The missions table is not rendering  as you can see in the caption below, kindly fix the issue:


Screenshot 2023-10-13 at 8 07 19 PM

**Review by algerina — CHANGES_REQUESTED** [review]
Hi @team,

Great job so far! However you have some minor issues to work on.

May I highlight the following: 
-  Well done working on your the Rockets and Dragons  pages and functionalities✔
- Your code is well organized✔️



## CHANGES REQUIRED:muscle::clap:


Check the comments under the review.



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

**Comment by rmiyoyo** [comment]
# Work Done After Review
- Fixed the issue of missions not rendering.
- Ran linter checks.

**Review by IngDavidTM — APPROVED** [review]
Hi @rmiyoyo 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**